#! /usr/bin/env python
# All the available plugins shorted by performance:
# kinematics_solver: grips_arm_kinematics/IKFastTransform6dPlugin
# kinematics_solver: lma_kinematics_plugin/LMAKinematicsPlugin
# kinematics_solver: grips_arm_kinematics/IterativeDecouplingPlugin
# kinematics_solver: kdl_kinematics_plugin/KDLKinematicsPlugin
PACKAGE = "moveit_kinematics_interface"

from dynamic_reconfigure.parameter_generator_catkin import *
from math import pi


gen = ParameterGenerator()

solvers_enum = gen.enum([ gen.const("IKFastTransform6dPlugin", int_t, 0, "IKFastTransform6dPlugin"),
                          gen.const("LMAKinematicsPlugin", int_t, 1, "LMAKinematicsPlugin"),
                          gen.const("IterativeDecouplingPlugin", int_t, 2, "IterativeDecouplingPlugin"),
                          gen.const("KDLKinematicsPlugin", int_t, 3, "KDLKinematicsPlugin") ],
                          "Available kinematics solvers")

#       Name                 Type   Reconfig. Level   Description                     Default    Min   Max
gen.add("kinematics_solver",  int_t, 0,               "Available kinematics solvers", 0,         0,    3, edit_method=solvers_enum)
gen.add("max_increment",   double_t, 0,               "Max angle inc. (Iterative)",   0.05,      1e-3, 2*pi)

exit(gen.generate(PACKAGE, "moveit_kinematics_interface", "KinematicsSolvers"))
